pkg/gadgets/run/tracer/run.go

```diff
@@ -550,6 +595,49 @@ func (g *GadgetDesc) getColumns(info *types.GadgetInfo) (*columns.Columns[types.
 	if err := cols.AddFields(fields, base); err != nil {
 		return nil, fmt.Errorf("adding fields: %w", err)
 	}
+
+	btfSpec, err := btf.LoadKernelSpec()
+	if err != nil {
+		return nil, fmt.Errorf("getting kernel BTF: %w", err)
+	}
+
+	// Converts enum values to their string representation
+	for _, e := range enums {
+		c, ok := cols.GetColumn(e.attrs.Name + "_raw")
+		if !ok {
+			continue
+		}
+		var getter func(*types.Event) uint64
+		if e.btfEnum.Signed {
+			getterT := columns.GetFieldAsNumberFunc[int64, types.Event](c)
+			getter = func(ev *types.Event) uint64 {
+				return uint64(getterT(ev))
+			}
+		} else {
+			getter = columns.GetFieldAsNumberFunc[uint64, types.Event](c)
+		}
+		cols.MustAddColumn(e.attrs, func(ev *types.Event) any {
+			if ev == nil || len(ev.RawData) == 0 {
+				return ""
+			}
+			val := getter(ev)
+
+			enum := &btf.Enum{}
+			err = btfSpec.TypeByName(e.btfEnum.Name, &enum)
+			if err != nil {
+				return "NOT_FOUND"
+			}
```

Comment:
This works fine if the enum involved is defined in the kernel, but what about if this is an enum defined by the user?

Reply:
Also, I think this logic can be moved outside this function to avoid calling it each time the conversion is done.

Reply:
Finally, what about adding a comment explaining why we use the kernel type to get the string representation?

Reply:
> For the moment, I kept this code in this function as I do not see where to put it.

I meant, to move the `btfSpec.TypeByName`  call outside the anonymous functions that is passed to `MustAddColumn`. That logic can be done just once instead of repeating it each time the extractor is called.